### PR TITLE
update ruby to 3.1.2 to match mac ventura

### PR DESCRIPTION
**Pull request by grosser**
... and will give us a solid speed boost + prep us for the future

/cc @zdrve

### CODE REVIEW

File: lib/kennel/utils.rb [hunk 1]
```diff
@@ -5,6 +5,7 @@ module Utils
 
     class TeeIO < IO
       def initialize(ios)
+        super(0)
```
Comment (zdrve) [inline lib/kennel/utils.rb:8]: That's an unfamiliar construct and worth a code comment, I think. Why 0?

Comment (grosser) [reply lib/kennel/utils.rb:8]: ah yeah ... a bogus value to make it stop complaining :D
... I'll add a comment

Comment (grosser) [reply lib/kennel/utils.rb]: updated

File: lib/kennel/utils.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ module Utils
 
     class TeeIO < IO
       def initialize(ios)
+        super(0) # called with fake file descriptor 0, so we can call super and get a proper class
```
Comment (zdrve) [inline lib/kennel/utils.rb:8]: FD 0 isn't "fake", it's stdin. Does it _matter_ that it's stdin? Would 1 (stdout) work just as well? If not, why not? Does this even need to be a subclass of `IO`?

The code _might_ be fine; the comment leaves a lot of questions unanswered.

Comment (zdrve) [reply lib/kennel/utils.rb:8]: I'm also of course aware that `TeeIO` isn't a new class, but the point stands regardless.

**Comment by grosser** [comment]
It needs to inherit from Io to get all it's methods.
Not really the point of this pr, can be its own issue if you think it's
worth discussing more, it works since a while and has not made any trouble
so I would just leave it alone for now.

**Review by zdrve — APPROVED** [review]
I'm curious about the `IO` thing now but I'll ponder that outside of this PR :-) 

Anyway. I notice that currently, kennel-config and kennel-lib do not have matching `.ruby-version`s. Should we fix that? Is that idea of this PR that we update k-config to 3.1.2 also, so that all kennel users are running 3.1.2?

**Comment by grosser** [comment]
yes, both should be in sync
still need to confirm the exact ventura ruby version and it's rollout schedule 🤞

**Pull request merged**